### ISSUE TITLE
Stricter POSIX compliance

### ISSUE THREAD

**Issue by mcandre** [direnv/direnv#633]
**Is your feature request related to a problem? Please describe.**

direnv appears to be missing support for POSIX sh and related shells.

**Describe the solution you'd like**

Offer direnv installer/library in pure POSIX sh.

**Describe alternatives you've considered**

Becoming a COBOL monk.

**Additional context**

As a bonus, any gains in POSIX sh support, translate directly into better support for ksh users!

**Comment by zimbatm** [comment]
Since POSIX sh doesn't have a notion of a pre-prompt hook, it would be hard to provide the same user experience as on the other shells. That being said, we could still approximate that by overriding `cd` and other builtins.

Bash is always going to be a dependency of direnv though.

**Comment by zimbatm** [comment]
So if somebody is interested in this, go for it :)

**Comment by periish** [comment]
Couldn’t you achieve this by running a direnv command on every time PS1 is run? Most shells support subshells being run in PS1.

**Comment by zimbatm** [comment]
Yes, that could work I suppose. The only way to know for sure is to implement and try it out for a while.

**Comment by periish** [comment]
Alright, I’ll try it out then.

**Comment by raiguard** [comment]
Bump. I use [yash](https://magicant.github.io/yash/) and would love to be able to use this software with it. Inspiration could be taken from [zoxide](https://github.com/ajeetdsouza/zoxide), which hooks `$PS1` as periish suggested.
